src/components/collapsible-motion/collapsible-motion.js

```diff
@@ -17,144 +27,225 @@ const createClosingAnimation = height =>
-                    containerStyles: {
-                      ...containerStyles,
-                      ...animationStyle,
-                    },
-                    toggle: animationToggle,
-                    registerContentNode,
-                  });
-                }}
-              </ClassNames>
-            )}
-          </ToggleAnimation>
-        )}
-      </Collapsible>
-    );
+
+  return props.children({
+    animation,
+    containerStyles,
+    toggle: handleToggle,
+    registerContentNode: nodeRef,
+  });
+};
+
+ToggleAnimation.displayName = 'ToggleAnimation';
+ToggleAnimation.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  toggle: PropTypes.func.isRequired,
+  children: PropTypes.func.isRequired,
+};
+
+const CollapsibleMotion = props => {
+  const isControlledComponent = !isNil(props.isClosed);
+
+  if (isControlledComponent) {
+    return <ControlledCollapsibleMotion {...props} />;
   }
-}
+
+  return <UncontrolledCollapsibleMotion {...props} />;
+};
+
+const ControlledCollapsibleMotion = props => {
+  return (
+    <Collapsible
+      isClosed={props.isClosed}
+      isDefaultClosed={props.isDefaultClosed}
+      onToggle={props.onToggle}
+    >
+      {({ isOpen, toggle }) => (
+        <ToggleAnimation isOpen={isOpen} toggle={toggle}>
+          {({
+            animation,
+            containerStyles,
+            toggle: animationToggle,
+            registerContentNode,
+          }) => (
+            <ClassNames>
+              {({ css }) => {
+                let animationStyle = {};
+
+                if (animation) {
+                  // By calling `css`, emotion injects the required CSS into the document head.
+                  // eslint-disable-next-line no-unused-expressions
+                  css`
+                    animation: ${animation} 200ms forwards;
+                  `;
+                  animationStyle = {
+                    animation: `${animation.name} 200ms forwards`,
+                  };
+                }
+
+                return props.children({
+                  isOpen,
+                  containerStyles: {
+                    ...containerStyles,
+                    ...animationStyle,
+                  },
+                  toggle: animationToggle,
+                  registerContentNode,
+                });
+              }}
+            </ClassNames>
+          )}
+        </ToggleAnimation>
+      )}
+    </Collapsible>
+  );
+};
+
+ControlledCollapsibleMotion.displayName = 'ControlledCollapsibleMotion';
+ControlledCollapsibleMotion.propTypes = {
+  children: PropTypes.func.isRequired,
+  isClosed: PropTypes.bool,
+  onToggle: PropTypes.func,
+  isDefaultClosed: PropTypes.bool,
+};
+
+const UncontrolledCollapsibleMotion = props => {
+  const [isOpen, toggle] = useToggleState(!props.isDefaultClosed);
+
+  const [
+    animation,
+    containerStyles,
+    animationToggle,
+    registerContentNode,
+  ] = useToggleAnimation(isOpen, toggle);
+
+  return (
+    <ClassNames>
```

Comment:
I wish there was a `useClassNames` hook  😢

Reply:
Wait a bit :)

Reply:
I think it's not that simple 😢 
https://github.com/emotion-js/emotion/issues/1321